features/filesystem/storagelite/source/StorageLite.h

```diff
@@ -0,0 +1,559 @@
+                       size_t& actual_file_name_size, uint32_t& handle);
+
+    /**
+     * @brief Returns name of next file given iteration handle (handle is updated).
+     *        Function return is STORAGELITE_NOT_FOUND at the end of the search.
+     *
+     * @param[in]  file_name             File name buffer.
+     * @param[in]  file_name_buf_size    File name buffer size.
+     * @param[out] actual_file_name_size Actual file name size.
+     * @param[out] handle                Iteration handle.
+     *
+     * @returns STORAGELITE_SUCCESS           Success.
+     *          STORAGELITE_NOT_FOUND         File was not found (past last file).
+     *          STORAGELITE_READ_ERROR        Physical error reading data.
+     *          STORAGELITE_DATA_CORRUPT      Data is corrupt.
+     *          STORAGELITE_BAD_VALUE         Bad value in any of the parameters.
+     *          STORAGELITE_BUFF_TOO_SMALL    Not enough memory in user buffer.
+     */
+    int get_next_file(uint8_t *file_name, size_t file_name_buf_size,
+                      size_t& actual_file_name_size, uint32_t& handle);
+
+    /**
+     * @brief Initializes StorageLite component.
+     *
+     * @param[in]  bd                     Underlying block device.
+     * @param[in]  max_files              Maximal number of files.
+     *
+     * @returns STORAGELITE_SUCCESS       Initialization completed successfully.
+     *          STORAGELITE_READ_ERROR    Physical error reading data.
+     *          STORAGELITE_WRITE_ERROR   Physical error writing data (on recovery).
+     */
+    int init(BlockDevice *bd, size_t max_files = STORAGELITE_MAX_FILES);
+
+    /**
+     * @brief Deinitializes StorageLite component.
+     *        Warning: This function is not thread safe and should not be called
+     *        concurrently with other StorageLite functions.
+     *
+     * @returns STORAGELITE_SUCCESS       Deinitialization completed successfully.
+     */
+    int deinit();
+
+    /**
+     * @brief Reset StorageLite areas (delete all data).
+     *
+     * @returns STORAGELITE_SUCCESS       Reset completed successfully.
+     *          STORAGELITE_READ_ERROR    Physical error reading data.
+     *          STORAGELITE_WRITE_ERROR   Physical error writing data.
+     */
+    int reset();
+
+    /**
+     * @brief Reset StorageLite to factory defaults. i.e. copy all files defined as
+     *        backup copies on top of their active copies.
+     *
+     * @returns STORAGELITE_SUCCESS       Reset completed successfully.
+     *          STORAGELITE_READ_ERROR    Physical error reading data.
+     *          STORAGELITE_WRITE_ERROR   Physical error writing data.
+     */
+    int factory_reset();
+
+    /**
+     * @brief Return StorageLite size (area size).
+     *
+     * @returns StorageLite size.
+     */
+    size_t size() const;
+
+    /**
+     * @brief Return StorageLite free size.
+     *
+     * @returns StorageLite free size.
+     */
+    size_t free_size() const;
```

Comment:
`free_size` is based on the current pointer to the next written record in the BD. One can't calculate it from the BD size (which is static). I thought 
Regarding `statvfs` - yes, was wondering whether to implement it or not. Most of the info in this structure has no meaning for StorageLite (like "Number of free blocks for unprivileged users"). Maybe will implement it anyway. No big deal.

Reply:
Oh interesting, is there a way to get the theoretical size if a gc were to happen?

I was thinking this was a representation of how "full" the filesystem was. Maybe these APIs are too confusing? Are they useful for something?

Reply:
It is used for that purpose ("how full"). Of course it's misleading, because it doesn't take future metadata and padding into account, just the "raw" free size.

Reply:
Anyway, `statvfs` has been implemented.